lib/chef/knife/ssh.rb

```diff
@@ -212,14 +234,24 @@ def search_nodes
 
         separator = ui.presenter.attribute_field_separator
 
+        # if we've set a prefix attribute to use on the command line
+        if config[:prefix]
+          required_attributes[:prefix_cli] = config[:prefix].split(separator)
+        end
+
+        # if we've configured a prefix attribute in our config
+        if Chef::Config[:knife][:prefix_attribute]
+          required_attributes[:prefix_config] = Chef::Config[:knife][:prefix_attribute].split(separator)
+        end
```

Comment:
again this should just be `config[:prefix_attribute]` and let the knife cli/config merging code handle it.